### PR TITLE
CI: add Github Action workflow for MacOS and Windows

### PR DESCRIPTION
**Pull request by Julow**
It builds and runs the tests on macos, ubuntu and windows and also run the `@help` alias.
It doesn't try different version of OCaml because that's already done by OCaml-CI.

### CODE REVIEW

**Comment by Julow** [comment]
The new check shows that the helps files weren't uptodate and that the tests fail on Windows (due to end of lines perhaps?)

**Review by 1972chris — APPROVED** [review]

File: test/cli/conf.t [hunk 1]
```diff
@@ -5,39 +5,39 @@ Invalid version:
 
 Exit code is printed by hand because sed succeeding would hide the error.
 
-  $ (<a.ml ocamlformat --impl -; echo [$?]) 2>&1 | sed 's/version is "[^"]*"/version is "..."/g'
-  ocamlformat: Error while parsing $TESTCASE_ROOT/.ocamlformat:
+  $ (<a.ml ocamlformat --impl -; echo [$?]) 2>&1 | sed 's/version is "[^"]*"/version is "..."/g' | make_deterministic
+  ocamlformat: Error while parsing <testcase_root>/.ocamlformat:
```
Comment (Julow) [inline test/cli/conf.t:9]: Why is this needed ?

Comment (gpetiot) [reply test/cli/conf.t:9]: Because dune doesn't pick up Windows paths and doesn't do the rewrite with TESTCASE_ROOT

Comment (Julow) [reply test/cli/conf.t:9]: Perhaps we should disable all the tests on Windows ? We could add a windows-specific formatting test to be notified of formatting changes and that would be already better than before.

Comment (gpetiot) [reply test/cli/conf.t]: Done, I've also added some tests for the Eol_compat lib. Everything looks good to me now.

**Review by gpetiot — APPROVED** [review]
@Julow if it looks good to you we can merge!

File: test/unit/test_indent.ml [hunk 2]
```diff
@@ -24,11 +24,13 @@ let x =
 y
 in
 2|}
-        ~expected:{|let f =
+        ~expected:
+          (Eol_compat.normalize_eol ~line_endings:`Lf
```
Comment (Julow) [inline test/unit/test_indent.ml:28]: Would be better inside the `test` function.

File: vendor/eol-compat/dune [hunk 3]
```diff
@@ -0,0 +1,3 @@
+(library
+ (name eol_compat)
```
Comment (Julow) [inline vendor/eol-compat/dune:2]: Why is this placed in the `vendor` directory ? Would be better in `lib`.

**Review by Julow — COMMENTED** [review]
I have a small comment but otherwise looks good.

File: test/unit/test_indent.ml (hunk 2 above)
Comment (gpetiot) [reply test/unit/test_indent.ml]: done

File: vendor/eol-compat/dune (hunk 3 above)
Comment (gpetiot) [reply vendor/eol-compat/dune]: done
